examples/named_entity_recognition/ner_wikigold_transformer.ipynb

```diff
@@ -0,0 +1,318 @@
+   "outputs": [],
+   "source": [
+    "## Set QUICK_RUN = True to run the notebook on a small subset of data and a smaller number of epochs.\n",
+    "QUICK_RUN = False"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Summary\n",
+    "\n",
+    "This notebook demonstrates how to fine tune [pretrained Transformer model](https://github.com/huggingface/transformers) for named entity recognition (NER) task. Utility functions and classes in the NLP Best Practices repo are used to facilitate data preprocessing, model training, model scoring, and model evaluation. \n",
+    "\n",
+    "The pretrained transformer of [BERT (Bidirectional Transformers for Language Understanding)](https://arxiv.org/pdf/1810.04805.pdf) architecture is used in this notebook. [BERT](https://arxiv.org/pdf/1810.04805.pdf) is a powerful pre-trained lanaguage model that can be used for multiple NLP tasks, including text classification, question answering, named entity recognition, etc. It's able to achieve state of the art performance with only a few epochs of fine tuning on task specific datasets.\n",
+    "\n",
+    "The figure below illustrates how BERT can be fine tuned for NER tasks. The input data is a list of tokens representing a sentence. In the training data, each token has an entity label. After fine tuning, the model predicts an entity label for each token in a given testing sentence. \n",
+    "\n",
+    "<img src=\"https://nlpbp.blob.core.windows.net/images/bert_architecture.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "import os\n",
+    "\n",
+    "nlp_path = os.path.abspath('../../')\n",
+    "if nlp_path not in sys.path:\n",
+    "    sys.path.insert(0, nlp_path)\n",
+    "\n",
+    "from utils_nlp.dataset import wikigold\n",
+    "from utils_nlp.common.timer import Timer\n",
+    "from seqeval.metrics import classification_report\n",
+    "from utils_nlp.models.transformers.named_entity_recognition import *"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Configuration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# fraction of the dataset used for testing\n",
+    "TEST_DATA_FRACTION = 0.3\n",
+    "\n",
+    "# sub-sampling ratio for training\n",
+    "TRAIN_SAMPLE_RATIO = 1\n",
+    "\n",
+    "# sub-sampling ratio for testing\n",
+    "TEST_SAMPLE_RATIO = 1\n",
+    "\n",
+    "# update variables for quick run option\n",
+    "if QUICK_RUN:\n",
+    "    TRAIN_SAMPLE_RATIO = 0.1\n",
+    "    TEST_SAMPLE_RATIO = 0.1\n",
+    "    NUM_TRAIN_EPOCHS = 1\n",
+    "\n",
+    "# the data path used to save the downloaded data file\n",
+    "DATA_PATH = \"./temp\"\n",
```

Comment:
We're replacing these temp paths with TemporaryDirectory
https://github.com/microsoft/nlp/blob/ee67aa0108a6e1928a068c66227242f94ef8c3a4/examples/text_classification/tc_mnli_transformers.ipynb